lib/collaborative_filtering.py

```diff
@@ -2,3 +2,212 @@
 """
 Module that provides the main functionalities of collaborative filtering.
 """
+
+from util.data_parser import DataParser
+from lib.evaluator import Evaluator
+from lib.abstract_recommender import AbstractRecommender
+from numpy.linalg import solve
+import numpy
+import sys
+import os
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
```

Comment:
I'm not comfortable having this line here, I would prefer if it's in the main(just for testing), since we shouldn't depend on global statements.